src/util/animate_color.ts

```diff
@@ -46,40 +62,46 @@ const defaultColorEasing = (currentTime, duration) =>
  * @returns {Function} abort function
  */
 export function animateColor(
-  fromColor,
-  toColor,
+  fromColor: string,
+  toColor: string,
   duration = 500,
   {
     colorEasing = defaultColorEasing,
     onComplete,
     onChange,
-    ...restOfOptions
+    abort,
+    target,
+  }: {
+    colorEasing?: ColorEasing;
+    onComplete?: OnColorChangeCallback;
+    onChange?: OnColorChangeCallback;
+    abort?: AbortCallback;
+    target?: unknown;
   } = {}
-) {
+): CancelFunction {
   const startColor = new Color(fromColor).getSource(),
     endColor = new Color(toColor).getSource();
   return animate({
-    ...restOfOptions,
+    target,
     duration,
+    abort,
     startValue: startColor,
     endValue: endColor,
     byValue: endColor,
     easing: (currentTime, startValue, byValue, duration) =>
-      calculateColor(startValue, byValue, colorEasing(currentTime, duration)),
+      colorEasing(currentTime, duration),
```

Comment:
this is a change, not just types

I am fine with it as it aligns with animation core

Reply:
Not in a typing pr and not with a clear explanation of what is changing and how.